### PR TITLE
feat: bootstrapping lists CPT

### PR DESCRIPTION
**Pull request by leogermani**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Creates a CPT to manage Subscription Lists

### How to test the changes in this Pull Request:

1. Check the code
2. Verify the Lists menu added under Newsletters
3. Try adding/editing Lists
4. Verify the content only shows a simple textarea

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by dkoo — CHANGES_REQUESTED** [review]
This works as described:

Screen Shot 2022-11-08 at 4 08 34 PM

However, when running `npm ci` I get an error that I don't see while on `master`:


```
npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
```

I don't see any changes to `package.json` or the lockfile, so not sure if it's related or if this branch just needs to be updated.

File: includes/class-subscription-lists.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Newspack Subscription Lists
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Newsletter;
```
Comment (miguelpeixe) [inline includes/class-subscription-lists.php:8]:
```suggestion
namespace Newspack\Newsletters;
```

The plugin is called "Newsletters"

File: includes/class-subscription-lists.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Newspack Subscription Lists
```
Comment (miguelpeixe) [inline includes/class-subscription-lists.php:3]:
```suggestion
 * Newspack Newsletters Subscription Lists
```

File: includes/class-newspack-newsletters.php [hunk 3]
```diff
@@ -962,7 +962,8 @@ public static function branding_scripts() {
 		$screen = get_current_screen();
 		if (
 			self::NEWSPACK_NEWSLETTERS_CPT !== $screen->post_type &&
-			Newspack_Newsletters_Ads::NEWSPACK_NEWSLETTERS_ADS_CPT !== $screen->post_type
+			Newspack_Newsletters_Ads::NEWSPACK_NEWSLETTERS_ADS_CPT !== $screen->post_type &&
+			Newspack\Newsletter\Subscription_Lists::NEWSPACK_NEWSLETTERS_LIST_CPT !== $screen->post_type
```
Comment (miguelpeixe) [inline includes/class-newspack-newsletters.php:966]:
```suggestion
			Newspack\Newsletters\Subscription_Lists::NEWSPACK_NEWSLETTERS_LIST_CPT !== $screen->post_type
```

File: includes/class-newspack-newsletters.php [hunk 4]
```diff
@@ -962,7 +962,8 @@ public static function branding_scripts() {
 		$screen = get_current_screen();
 		if (
 			self::NEWSPACK_NEWSLETTERS_CPT !== $screen->post_type &&
-			Newspack_Newsletters_Ads::NEWSPACK_NEWSLETTERS_ADS_CPT !== $screen->post_type
+			Newspack_Newsletters_Ads::NEWSPACK_NEWSLETTERS_ADS_CPT !== $screen->post_type &&
```
Comment (miguelpeixe) [inline includes/class-newspack-newsletters.php:965]: Will the namespace change be made in another PR?

File: includes/class-subscription-lists.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Newsletter;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main Newspack Subscription Lists class.
```
Comment (miguelpeixe) [inline includes/class-subscription-lists.php:13]:
```suggestion
 * Main Newspack Newsletters Subscription Lists class.
```

File: includes/class-subscription-lists.php [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+ * Subscriptions Lists are Lists which readers can subscribe to. AKA Newsletters.
+ *
+ * Each List is associated with a Audience/List in the Provider and can be associated to one or more tags in the provider
+ */
+class Subscription_Lists {
+
+	/**
+	 * CPT for Newsletter Lists.
+	 */
+	const NEWSPACK_NEWSLETTERS_LIST_CPT = 'newsletter_list';
```
Comment (miguelpeixe) [inline includes/class-subscription-lists.php:24]: We should use a proper plugin prefix for this. Perhaps `newspack_newsletters_list`?

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
The npm issue that @dkoo mentioned was resolved with 6947a698063cce879d5959cd3265563f1da3820b

The CPT is working well, some further discussion is needed in the comments below.

File: includes/class-newspack-newsletters.php (hunk 4 above)
Comment (leogermani) [reply includes/class-newspack-newsletters.php:965]: what namespace change? In Ads?

File: includes/class-subscription-lists.php (hunk 6 above)
Comment (leogermani) [reply includes/class-subscription-lists.php]: :+1: done

File: includes/class-newspack-newsletters.php (hunk 4 above)
Comment (miguelpeixe) [reply includes/class-newspack-newsletters.php:965]: This is an ad feature inside Newsletters (`includes/class-newspack-newsletters-ads.php`). I assume the new namespace will be applied to all of the plugin classes eventually, correct?

File: includes/class-subscription-lists.php (hunk 6 above)
Comment (leogermani) [reply includes/class-subscription-lists.php]: Actually no, post type names shouldn't be longer than 20 characters

File: includes/class-newspack-newsletters.php (hunk 4 above)
Comment (leogermani) [reply includes/class-newspack-newsletters.php:965]: I'd love if we started using more namespaces across the codebase... and also that we relied on the autoloader so we don't have to keep `require_once` everything. But I figured this is out of the scope of this PR and this project... let's do baby steps.

BUT, what I considered doing there was to not have all these `if`s there. Instead, each of the 3 classes would enqueue the same script, each one in its own page... so we don't have this cross reference. There's a better hook to use. I could do that

File: includes/class-subscription-lists.php (hunk 6 above)
Comment (miguelpeixe) [reply includes/class-subscription-lists.php]: The newsletters post type is called `newspack_nl_cpt`. We could borrow the `_nl_` and make `newspack_nl_list`?

Comment (leogermani) [reply includes/class-subscription-lists.php]: yep. done

File: newspack-newsletters.php [hunk 7]
```diff
@@ -55,3 +55,5 @@
 require_once NEWSPACK_NEWSLETTERS_PLUGIN_FILE . '/includes/class-newspack-newsletters-quick-edit.php';
 require_once NEWSPACK_NEWSLETTERS_PLUGIN_FILE . '/includes/class-newspack-newsletters-embed.php';
 require_once NEWSPACK_NEWSLETTERS_PLUGIN_FILE . '/includes/class-newspack-newsletters.php';
+
+\Newspack\Newsletters\Subscription_Lists::init();
```
Comment (miguelpeixe) [inline newspack-newsletters.php:59]: Until we fully change the strategy on how things are required and initialized, can we keep the `init()` inside the class file as we do for all other classes in the plugin?

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: newspack-newsletters.php (hunk 7 above)
Comment (leogermani) [reply newspack-newsletters.php:59]: Any particular reason for that?

I think we should start moving forward doing this the "right" way... meaning relying on the autoloader (so the classes are onlyu loaded if needed) and initializing things from outside the file. But a huge refactor might be traumatic, so we can so it step by step...

**Review by miguelpeixe — APPROVED** [review]
